File: Backend/Game Buddy Finder/Game Buddy Finder/Models/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Game_Buddy_Finder.Models
{
    public class Message
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int MessageId { get; set; }

        [Required]
        public DateTime SendTime { get; set; }

        [Required]
        public int SenderId { get; set; }

        [Required]
        public int ReceiverId { get; set; }

        [Required]
        public string Subject { get; set; }

        [Required]
        public string Content { get; set; }

        [NotMapped]
        public string SenderUsername { get; set; }

        [NotMapped]
        public string ReceiverUsername { get; set; }

    }
}

File: Frontend/game-buddy-finder/src/app/models/message.ts
export class Message {
    messageId: number;
    sendTime;
    senderId: number;
    receiverId: number;
    subject: string;
    content: string;
    receiverUsername: string = "";

    public constructor(senderId: number, receiverId: number, subject: string, content: string) {
        this.senderId = senderId;
        this.receiverId = receiverId;
        this.subject = subject;
        this.content = content;
    }
}
